### PR TITLE
Intl refactor for new Top Up components

### PR DESCRIPTION
**Pull request by wackerow**
Intl refactoring for the new Top Up components
- TopupPage.tsx
- TopUpTransactionModal.tsx
- TransactionProgress.tsx
- ValidatorTable.tsx
- WalletConnectModal.tsx
- pages/TopUp/index.tsx
- AppBar.tsx

@CarlBeek Am I correct that the strings in `transactionErrorTypes.ts` are checking for different types of rejection errors, and should not be translated?

Also fixes https://github.com/ethereum/eth2.0-deposit/issues/368

### CODE REVIEW

**Review by scott1979rn — APPROVED** [review]

**Review by scott1979rn — APPROVED** [review]

File: src/pages/TopUp/index.tsx [hunk 1]
```diff
@@ -138,8 +141,7 @@ const _TopUpPage: React.FC<Props> = () => {
           <div style={{ display: 'flex', justifyContent: 'center' }}>
             <AlertIcon color="redLight" />
             <Text className="ml10">
-              There was an issue loading your validator information from
-              Beaconcha.in
+              <FormattedMessage defaultMessage="There was an issue loading your validator information from Beaconcha.in" />
```
Comment (ryancreatescopy) [inline src/pages/TopUp/index.tsx:144]: @CarlBeek what is the suggested user action if they see this?

File: src/pages/TopUp/components/TransactionProgress.tsx [hunk 2]
```diff
@@ -111,11 +112,15 @@ const TransactionProgress: React.FC<TransactionProgressProps> = ({
...
-        <Text className="mt10 ml15">Sign transaction with your wallet</Text>
+        <Text className="mt10 ml15">
+          <FormattedMessage defaultMessage="Sign transaction with your wallet" />
+        </Text>
       </div>
       <div className="mt30 flex">
         <ProgressCircle step={2} status={confirmOnChainStatus} />
-        <Text className="mt10 ml15">Confirm on-chain</Text>
+        <Text className="mt10 ml15">
+          <FormattedMessage defaultMessage="Confirm on-chain" />
```
Comment (ryancreatescopy) [inline src/pages/TopUp/components/TransactionProgress.tsx:122]: @CarlBeek what does this mean?

File: src/pages/TopUp/components/TopupPage.tsx [hunk 3]
```diff
@@ -181,20 +184,37 @@ const TopupPage: React.FC<Props> = ({ validator }) => {
...
+        defaultMessage:
+          'Validator balance is already higher than the maximum amount.',
+      });
     if (balanceAfterTopup > 32)
-      return `Submitting a topup for more than ${maxTopUpVal.toFixed(
-        4
-      )} will result in a balance higher than the maximum balance of 32.`;
+      return formatMessage(
+        {
+          defaultMessage: `Validators have a maximum balance of 32 ETH. You only need to top up {maxTopUpValue}.`,
```
Comment (ryancreatescopy) [inline src/pages/TopUp/components/TopupPage.tsx:194]: @wackerow we should add a PRICE_PER_VALIDATOR variable to this string IMO

**Comment by ryancreatescopy** [comment]
@wackerow @CarlBeek @samajammin been through the top up stuff. Quite hard to QA this right now. Any ideas how we can get it so that the app thinks we have validators to top up? I can't seem to figure out how...

**Comment by ryancreatescopy** [comment]
image

thoughts on layout for the nav bar? (App actions on the right, web page actions on the left)

**Comment by samajammin** [comment]
@ryancreatescopy happy to pair & show you but essentially you'll want to replace the `account` variable on this line w/ an address that has validators:
https://github.com/ethereum/eth2.0-deposit/blob/intlTopUp/src/pages/TopUp/index.tsx#L70

So e.g. replace
```
fetch(`${BEACONCHAIN_URL}/api/v1/validator/eth1/${account}`)
```

With 
```
fetch(`${BEACONCHAIN_URL}/api/v1/validator/eth1/0xa40dFEE99E1C85DC97Fdc594b16A460717838703`)
```

Where the address is an account that's deposited in the Eth2 deposity contract:
https://etherscan.io/address/0xa40dfee99e1c85dc97fdc594b16a460717838703

File: src/pages/TopUp/components/TopupPage.tsx (hunk 3 above)
Comment (ryancreatescopy) [reply src/pages/TopUp/components/TopupPage.tsx:194]: @wackerow I also changed the copy a fair bit here so worth a double check.

File: src/pages/TopUp/components/TransactionProgress.tsx (hunk 2 above)
Comment (samajammin) [reply src/pages/TopUp/components/TransactionProgress.tsx:122]: I'm guessing "Confirm the transaction on-chain", e.g. via a block explorer? Or is this a separate tx?

File: src/pages/TopUp/components/WalletConnectModal.tsx [hunk 4]
```diff
@@ -50,15 +51,20 @@ const WalletConnectModal: React.FC = () => {
...
             size="xlarge"
             style={{ margin: '45 auto', display: 'block' }}
           />
           <Text center>
-            Please connect to{' '}
-            {IS_MAINNET ? 'Ethereum Mainnet' : 'Göerli Testnet'}
+            <FormattedMessage
+              defaultMessage="Connect to {network}"
+              values={{
+                network: IS_MAINNET ? 'Ethereum Mainnet' : 'Göerli Testnet',
```
Comment (samajammin) [inline src/pages/TopUp/components/WalletConnectModal.tsx:64]:
```suggestion
                network: IS_MAINNET ? 'Ethereum mainnet' : 'Göerli testnet',
```

File: src/pages/TopUp/components/WalletConnectModal.tsx [hunk 5]
```diff
@@ -50,15 +51,20 @@ const WalletConnectModal: React.FC = () => {
...
+              }}
+              description="{network} is either 'Ethereum Mainnet' or 'Göerli Testnet'"
```
Comment (samajammin) [inline src/pages/TopUp/components/WalletConnectModal.tsx:66]:
```suggestion
              description="{network} is either 'Ethereum mainnet' or 'Göerli testnet'"
```

**Review by samajammin — APPROVED** [review]
Made a couple small changes. LGTM!

**Comment by samajammin** [comment]
LGTM but waiting on questions for @CarlBeek before merging.

**Comment by ryancreatescopy** [comment]
@samajammin @wackerow added a small message about effective balance and linked out to that article. I think it's useful but understand if we think it's too much.

image

File: src/pages/TopUp/index.tsx [hunk 6]
```diff
@@ -184,29 +193,40 @@ const _TopUpPage: React.FC<Props> = () => {
...
-            validator rewards are based on effective balance which is capped at
-            32 ETH.
+            <FormattedMessage
+              defaultMessage="You may need to top up your validator's balance for two important reasons. If your validator's effective balance is below {PRICE_PER_VALIDATOR} ETH you won't be earning your full staker rewards. And if it drops as low as {EJECTION_PRICE} ETH the system will eject your validator."
+              values={{ PRICE_PER_VALIDATOR, EJECTION_PRICE }}
+            />
           </Text>
+          <Alert variant="info" className="my20">
+            <FormattedMessage
+              defaultMessage="{PRICE_PER_VALIDATOR} ETH is the maximum effective validator balance. This means you won't earn more rewards if your validator's effective balance goes above {PRICE_PER_VALIDATOR}. However you will earn less if it dips below {PRICE_PER_VALIDATOR}."
```
Comment (samajammin) [inline src/pages/TopUp/index.tsx:219]:
```suggestion
              defaultMessage="{PRICE_PER_VALIDATOR} ETH is the maximum effective validator balance. This means you won't earn more rewards if your validator's balance goes above {PRICE_PER_VALIDATOR}. However you will earn less if it dips below {PRICE_PER_VALIDATOR}."
```

Comment (samajammin) [reply src/pages/TopUp/index.tsx:219]: @ryancreatescopy slightly tweaked this. 

> This means you won't earn more rewards if your validator's effective balance goes above 32 ETH

It's not possible for the *effective* balance to go above 32 ETH.

**Review by samajammin — APPROVED** [review]

File: src/pages/TopUp/index.tsx (hunk 6 above)
Comment (ryancreatescopy) [reply src/pages/TopUp/index.tsx]: :+1:

**Pull request merged**